src/string_bytes.cc

```diff
@@ -700,6 +708,11 @@ Local<Value> StringBytes::Encode(Isolate* isolate,
   if (!buflen && encoding != BUFFER)
     return scope.Escape(String::Empty(isolate));
 
+  if (buflen > String::kMaxLength && encoding != BUFFER) {
+    ThrowRangeError(isolate, "invalid String length");
```

Comment:
Throwing a JS exception from here feels like too much side effect, that should only be done at the edges of the JS/C++ bridge.